src/file_io.jl

```diff
@@ -22,16 +24,32 @@ end
 # they will be `deserialized` when they are indexed out of the returned JSLOFile object.
 function Base.read(io::IO, ::Type{JLSOFile})
     d = BSON.load(io)
+    _versioncheck(d["metadata"]["version"], READABLE_VERSIONS)
+    upgrade_jlso!(d)
     return JLSOFile(
         d["metadata"]["version"],
         d["metadata"]["julia"],
         d["metadata"]["format"],
+        d["metadata"]["compression"],
         d["metadata"]["image"],
         d["metadata"]["pkgs"],
         d["objects"],
     )
 end
 
+function upgrade_jlso!(raw_dict::AbstractDict)
+    metadata = raw_dict["metadata"]
+    if metadata["version"] ∈ semver_spec("1")
+        if metadata["format"] == :serialize
```

Comment:
```suggestion
        if metadata["format"] === :serialize
```

Reply:
This has literally no effect.

Reply:
interesting. why do we `===` for Symbols all over the place?

Reply:
Sometimes (and possibly only for some julia versions)
that would allow it to specialize out the branch if there are non-Symbol options.
In theory.

My belief is now days, one should use `==` when you mean equality,
and let the compiler work out what do do.
It should trivailly inline into `===`, where it matters.

In this particular case, it does not matter.

Reply:
I think we should be using `===` for `Symbol`s. It can't produce _worse_ performance than `==` since the compiler loves `===`, it can only be the same or better.

Reply:
I guess so.